### PR TITLE
Edit bug

### PR DESCRIPTION
**Pull request by mimizotti**
In this branch, I:

- Implemented @kat3kasper's refactoring suggestion for the `editFood(input, newFoodValue, currentFoodValue, objectId, thisObject)` function
- Included `event` in the `validateFoodInput(name, calories, event)` so the function can have access to this variable
- Implemented @swdonovan request to validate the numeric value of calories (I threw in `(isNaN(calories))` in `static validateFoodInput(name, calories)`
- Fixed the edit bug: When you edit calories, the `content-editable` field no longer includes the delete button
- Fixed the delete bug: When you delete a food item, the page no longer refreshes.

### CODE REVIEW

**Comment by swdonovan** [comment]
this is awesome and all and i really like it. but we need to now have you work on this thing they call sleep.  origins came about around the same time as jogging (the silent J kind) and has been trending since!  Nice work here mimi!!!

File: lib/event_listeners/foodEvents.js [hunk 1]
```diff
@@ -1,70 +1,55 @@
...
-      event.preventDefault();
-      $("p#submit-calories").append("Please enter a calorie amount.")
-    } else {
-      $("p#submit-food-name").remove()
-      $("p#submit-calories").remove()
-      Ajax.postFoods(name, calories)
-      $('input').val('').removeAttr('checked').removeAttr('selected');
-    }
+    let thisObject = event
+    return FoodHandler.validateFoodInput(name, calories, thisObject)
```
Comment (swdonovan) [inline lib/event_listeners/foodEvents.js:13]: nice

File: lib/event_listeners/foodEvents.js [hunk 2]
```diff
@@ -1,70 +1,55 @@
...
 
-$("#food-list").click(event => {
-  let objectId = event.target.parentElement.parentElement.dataset.id
-  let row = event.target.parentElement.parentElement.parentElement
-  Ajax.deleteFood(objectId, row)
+
+$(document).ready(function(){
+  $("#food-list").click(event => {
+    if ((event.target.className === "delete-button") || (event.target.parentElement.className === "delete-button")) {
+    let objectId = event.target.parentElement.parentElement.dataset.id
```
Comment (swdonovan) [inline lib/event_listeners/foodEvents.js:23]: I still think this is crazy how this is much faster than calling .parent(). dang built in methods

**Review by swdonovan — APPROVED** [review]
328 lines deleted??? nice refactor!

**Comment by kat3kasper** [comment]
@mimilettd I agree with @swdonovan make sure you are getting some sleep 😆

**Pull request merged**